lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,138 @@
+require 'ipaddr'
+require 'pathname'
+
+module LinuxAdmin
+  class NetworkInterfaceRH < NetworkInterface
+    IFACE_DIR = "/etc/sysconfig/network-scripts"
+
+    # @return [Hash<String, String>] Key value mappings in the interface file
+    attr_reader :interface_conf
+
+    # @param interface [String] Name of the network interface to manage
+    def initialize(interface)
+      super
+      @iface_file = Pathname.new(IFACE_DIR).join("ifcfg-#{@interface}")
+      parse_conf
+    end
+
+    # Parses the interface configuration file into the @interface_conf hash
+    def parse_conf
+      @interface_conf = {}
+
+      File.read(@iface_file).each_line do |line|
```

Comment:
You can do `File.foreach(@iface_file) do |line|`  Then you aren't reading the whole file into memory at once.